src/main/java/io/pravega/connectors/flink/FlinkPravegaInputFormat.java

```diff
@@ -0,0 +1,179 @@
+
+    // The supplied event deserializer.
+    private final DeserializationSchema<T> deserializationSchema;
+
+    // The pravega controller endpoint.
+    private final URI controllerURI;
+
+    // The scope name of the destination stream.
+    private final String scopeName;
+
+    // The readergroup name to coordinate the parallel readers. This should be unique for a Flink job.
+    private final String readerGroupName;
+
+    // The names of Pravega streams to read
+    private final Set<String> streamNames;
+
+    // The configured start time for the reader
+    private final long startTime;
+
+    // The event read timeout
+    private long eventReadTimeout = DEFAULT_EVENT_READ_TIMEOUT;
+
+    // The Pravega reader; a new reader will be opened for each input split
+    private transient EventStreamReader<T> pravegaReader;
+
+    // Read-ahead event; null indicates that end of input is reached
+    private transient EventRead<T> lastReadAheadEvent;
+
+    /**
+     * Creates a new Flink Pravega {@link InputFormat} which can be added as a source to a Flink batch job.
+     *
+     * <p>The number of created input splits is equivalent to the parallelism of the source. For each input split,
+     * a Pravega reader will be created to read from the specified Pravega streams. Each input split is closed when
+     * the next read event returns {@code null} on {@link EventRead#getEvent()}.
+     *
+     * @param controllerURI         The pravega controller endpoint address.
+     * @param scope                 The destination stream's scope name.
+     * @param streamNames           The list of stream names to read events from.
+     * @param startTime             The start time from when to read events from.
+     *                              Use 0 to read all stream events from the beginning.
+     * @param deserializationSchema The implementation to deserialize events from pravega streams.
+     */
+    public FlinkPravegaInputFormat(
+            final URI controllerURI,
+            final String scope,
+            final Set<String> streamNames,
+            final long startTime,
+            final DeserializationSchema<T> deserializationSchema) {
+
+        Preconditions.checkNotNull(controllerURI, "controllerURI");
+        Preconditions.checkNotNull(scope, "scope");
+        Preconditions.checkNotNull(streamNames, "streamNames");
+        Preconditions.checkArgument(startTime >= 0, "start time must be >= 0");
+        Preconditions.checkNotNull(deserializationSchema, "deserializationSchema");
+
+        this.controllerURI = controllerURI;
+        this.scopeName = scope;
+        this.deserializationSchema = deserializationSchema;
+        this.streamNames = streamNames;
+        this.startTime = startTime;
+        this.readerGroupName = generateRandomReaderGroupName();
+
+        // TODO: This will require the client to have access to the pravega controller and handle any temporary errors.
```

Comment:
As a general suggestion, I like the idea of creating issues for TODOs in the code so that we can track what we need to fix in the future.